src/Sylius/Behat/Context/Ui/Admin/ManagingExchangeRatesContext.php

```diff
@@ -0,0 +1,371 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Ui\Admin;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Page\Admin\Crud\IndexPageInterface;
+use Sylius\Behat\Page\Admin\ExchangeRate\CreatePageInterface;
+use Sylius\Behat\Page\Admin\ExchangeRate\UpdatePageInterface;
+use Sylius\Component\Currency\Model\ExchangeRateInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+final class ManagingExchangeRatesContext implements Context
+{
+    /**
+     * @var CreatePageInterface
+     */
+    private $createPage;
+
+    /**
+     * @var IndexPageInterface
+     */
+    private $indexPage;
+
+    /**
+     * @var UpdatePageInterface
+     */
+    private $updatePage;
+
+    /**
+     * @param CreatePageInterface $createPage
+     * @param IndexPageInterface $indexPage
+     * @param UpdatePageInterface $updatePage
+     */
+    public function __construct(
+        CreatePageInterface $createPage,
+        IndexPageInterface $indexPage,
+        UpdatePageInterface $updatePage
+    ) {
+        $this->createPage = $createPage;
+        $this->indexPage = $indexPage;
+        $this->updatePage = $updatePage;
+    }
+
+    /**
+     * @Given I want to add a new exchange rate
+     */
+    public function iWantToAddNewExchangeRate()
+    {
+        $this->createPage->open();
+    }
+
+    /**
+     * @Given /^I want to edit (this exchange rate)$/
+     * @When /^I am editing (this exchange rate)$/
+     */
+    public function iWantToEditThisExchangeRate(ExchangeRateInterface $exchangeRate)
+    {
+        $this->updatePage->open(['id' => $exchangeRate->getId()]);
+    }
+
+    /**
+     * @When I want to browse exchange rates of the store
+     */
+    public function iWantToBrowseExchangeRatesOfTheStore()
+    {
+        $this->indexPage->open();
+    }
+
+    /**
+     * @When /^I specify its ratio as (-?[0-9\.]+)$/
+     * @When I don't specify its ratio
+     */
+    public function iSpecifyItsRatioAs($ratio = null)
+    {
+        if (null !== $ratio) {
+            $this->createPage->specifyRatio($ratio);
```

Comment:
Shouldn't we specify ratio always and in case "we don't specify" put an empty string? I think we do it that way in all other cases, please check.